enhancements/sig-architecture/32-extensiblescheduling/32-extensiblescheduling.md

```diff
@@ -0,0 +1,362 @@
+- Placement filter will not support the new API(CRD).
+
+## Proposal
+
+### User Stories
+
+#### Story 1: Users could use the data pushed from each managed cluster to select clusters.
+  - On each managed cluster, there is a customized agent monitoring the resource usage (eg. CPU ratio) of the cluster. It will calculate a scalar value and push the result to hub.
+  - As an end user, I can configure placement yaml to use this scalar value to select clusters.
+  
+#### Story 2: Users could use the metrics collected on hub to select clusters.
+  - On the hub, there is a customized agent to get metrics (eg. cluster allocatable memory) from Thanos. It will generate a scalar value for each cluster.
+  - As an end user, I can configure placement yaml to use this scalar value to select clusters.
+
+#### Story 3: Disaster recovery workload could be automatically switched to an avaliable cluster.
+  - A user has two clusters, a primary and backup cluster on which workload storage is configured so storage are synced from primary cluster to backup.
+  - As an end user, I want to deploy workloads on the primary cluster first. And when primary cluster is down, the workload should be automatically switched to the backup cluster.
+
+#### Story 4: Users could use the data provided by third party controllers to select clusters.
+  - A third party controller can comprehensively evaluate cluster on latency, region, iops etc. and rate cluster according to the SLA the cluster can provide.
+  - As an end user, I can configure placement yaml to use this scalar value to select clusters.
+
+### Risks and Mitigation
+N/A
+
+## Design Details
+
+### ManagedClusterScalar API
+ManagedClusterScalar is the new API to add, to represents a scalable value (aka score) of one managed cluster.
+```go
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:scope="Namespaced"
+// +kubebuilder:subresource:status
+
+// ManagedClusterScalar represents a scalable value (aka score) of one managed cluster.
+// Each ManagedClusterScalar only represents the scalar value of one prioritizer.
+// ManagedClusterScalar is a namesapce scoped resource. The namespace of the resource is the cluster namespace.
+type ManagedClusterScalar struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the attributes of the ManagedClusterScalar.
+	// +kubebuilder:validation:Required
+	// +required
+	Spec ManagedClusterScalarSpec `json:"spec"`
+
+	// Status represents the status of the ManagedClusterScalar.
+	// +optional
+	Status ManagedClusterScalarStatus `json:"status,omitempty"`
+}
+
+// ManagedClusterScalarSpec defines the attributes of the ManagedClusterScalar.
+type ManagedClusterScalarSpec struct {
+	// PrioritizerName will be the prioritizer name used in placement.
+	// +kubebuilder:validation:Required
+	// +required
+	PrioritizerName string `json:"prioritizerName,omitempty"`
```

Comment:
Is it possible that a ManagedClusterScalar is used by two prioritizers?

Reply:
yes, in the current design a ManagedClusterScalar can be used by two prioritizers.

Reply:
Maybe not, the prioritizers plugin should check if the CR spec.PrioritizerName equals the prioritizers plugin name.